docs/sdk/src/reference_docs/frame_runtime_upgrades_and_migrations.rs

```diff
@@ -0,0 +1,135 @@
+//! 	MyCustomMigration,
+//! 	// ...more migrations here
+//! );
+//! pub type Executive = frame_executive::Executive<
+//! 	Runtime,
+//! 	Block,
+//! 	frame_system::ChainContext<Runtime>,
+//! 	Runtime,
+//! 	AllPalletsWithSystem,
+//! 	Migrations, // <-- pass your migrations to Executive here
+//! >;
+//! ```
+//!
+//! ### Ensuring Single Block Migration Safety
+//!
+//! "My migration unit tests pass, so it should be safe to deploy right?"
+//!
+//! No! Unit tests execute the migration in a very simple test environment, and cannot account
+//! for the complexities of a real runtime or real on-chain state.
+//!
+//! Prior to deploying migrations, it is critical to perform additional checks to ensure that when
+//! run in our real runtime they will not brick the chain due to:
+//! - Panicing
+//! - Touching too many storage keys and resulting in an excessively large PoV
+//! - Taking too long to execute
+//!
+//! [`try-runtime-cli`](https://github.com/paritytech/try-runtime-cli) has a sub-command
+//! [`on-runtime-upgrade`](https://paritytech.github.io/try-runtime-cli/try_runtime_core/commands/enum.Action.html#variant.OnRuntimeUpgrade)
+//! which is designed to help with exactly this.
+//!
+//! Developers MUST run this command before deploying migrations to ensure they will not
+//! inadvertently result in a bricked chain.
+//!
+//! It is recommended to run as part of your CI pipeline. See the
+//! [polkadot-sdk check-runtime-migration job](https://github.com/paritytech/polkadot-sdk/blob/4a293bc5a25be637c06ce950a34490706597615b/.gitlab/pipeline/check.yml#L103-L124)
+//! for an example of how to configure this.
+//!
+//! ### Note on the Manipulability of PoV Size and Execution Time
+//!
+//! While [`try-runtime-cli`](https://github.com/paritytech/try-runtime-cli) can help ensure with
+//! very high certianty that a migration will succeed given **existing** on-chain state, it cannot
+//! prevent a malicious actor from manipulating state in a way that will cause the migration to take
+//! longer or produce a PoV much larger than previously measured.
+//!
+//! Therefore, it is important to write migrations in such a way that the execution time or PoV size
+//! it adds to the block cannot be easily manipulated. e.g., in your migration, do not iterate over
+//! storage that can quickly or cheaply be bloated.
+//!
+//! If writing your migration in such a way is not possible, a multi block migration should be used
+//! instead.
+//!
+//! ### Other useful tools
+//!
+//! [`Chopsticks`](https://github.com/AcalaNetwork/chopsticks) is another tool in the Substrate
+//! ecosystem which developers may find useful to use in addition to `try-runtime-cli` when testing
+//! their single block migrations.
+//!
+//! ## Multi Block Migrations
+//!
+//! Safely and easily execute long-running migrations across multiple blocks.
+//!
+//! Suitable for migrations which could use arbitrary amounts of block weight.
+//!
+//! TODO: Link to multi block migration example/s.
```

Comment:
```suggestion
//! TODO: Link to multi block migration example/s.
//! https://github.com/paritytech/polkadot-sdk/pull/2119
```

There's this PR open